platform/osgi/platform-osgi-conditions/src/main/java/org/codice/ddf/condition/BundleNameCondition.java

```diff
@@ -0,0 +1,131 @@
+   */
+  @Override
+  public boolean isPostponed() {
+    return false;
+  }
+
+  /**
+   * Returns whether the Condition is satisfied. This method is only called for immediate Condition
+   * objects or immutable postponed conditions, and must always be called inside a permission check.
+   * Mutable postponed Condition objects will be called with the grouped version {@link
+   * #isSatisfied(Condition[],Dictionary)} at the end of the permission check.
+   *
+   * @return {@code true} to indicate the Conditions is satisfied. Otherwise, {@code false} if the
+   *     Condition is not satisfied.
+   */
+  @Override
+  public boolean isSatisfied() {
+    if (args.length == 0) {
+      return false;
+    }
+    String bundleName = bundle.getHeaders().get("Bundle-SymbolicName");
+    String key = bundleName + args[0];
+    Boolean storedResult = DECISION_MAP.get(key);
+    if (storedResult != null) {
+      return storedResult;
+    }
+    String[] bundles = REGEX.split(args[0]);
+    for (String bundleStr : bundles) {
+      if (bundleName.contains(bundleStr)) {
+        DECISION_MAP.put(key, true);
+        return true;
+      }
+    }
+    DECISION_MAP.put(key, false);
+    return false;
+  }
+
+  /**
+   * Returns whether the Condition is mutable. A Condition can go from mutable ({@code true}) to
+   * immutable ({@code false}) over time but never from immutable ({@code false}) to mutable ({@code
+   * true}).
+   *
+   * @return {@code true} {@link #isSatisfied()} can change. Otherwise, {@code false} if the value
+   *     returned by {@link #isSatisfied()} will not change for this condition.
+   */
+  @Override
+  public boolean isMutable() {
+    return false;
+  }
+
+  /**
+   * Returns whether the specified set of Condition objects are satisfied. Although this method is
+   * not static, it must be implemented as if it were static. All of the passed Condition objects
+   * will be of the same type and will correspond to the class type of the object on which this
+   * method is invoked. This method must be called inside a permission check only.
+   *
+   * @param conditions The array of Condition objects, which must all be of the same class and
+   *     mutable. The receiver must be one of those Condition objects.
+   * @param context A Dictionary object that implementors can use to track state. If this method is
+   *     invoked multiple times in the same permission check, the same Dictionary will be passed
+   *     multiple times. The SecurityManager treats this Dictionary as an opaque object and simply
+   *     creates an empty dictionary and passes it to subsequent invocations if multiple invocations
+   *     are needed.
+   * @return {@code true} if all the Condition objects are satisfied. Otherwise, {@code false} if
+   *     one of the Condition objects is not satisfied.
+   */
+  @Override
+  public boolean isSatisfied(Condition[] conditions, Dictionary<Object, Object> context) {
+    for (Condition condition : conditions) {
```

Comment:
:pencil: could be rewritten as `return Stream.of(conditions).allMatch(Condition::isSatisfied)`

Reply:
Going to leave Streaming out of this stuff since traditional loops are usually faster.